core/network/src/discovery.rs

```diff
@@ -365,6 +393,6 @@ mod tests {
 			}
 		});
 
-		tokio::runtime::Runtime::new().unwrap().block_on(fut).unwrap();
+		tokio::run(fut)
```

Comment:
`tokio::run` is equivalent to the code that was previously there, except that it silently ignores errors.

Let's put this back:

```suggestion
		tokio::runtime::Runtime::new().unwrap().block_on(fut).unwrap();
```

Reply:
I put it back because I don't mind, though I don't follow what you mean by "silently ignores errors". It seems to me that `tokio::run` takes a `Future<Item=(),Error=()>`, so it only compiles if the future does not produce any meaningful errors, which is the case here and if the error type would change, it would no longer compile. I thus don't see how errors would be unintentionally silenced by using `tokio::run`. :man_shrugging:

Reply:
I'm thinking of runtime errors.
If the `Future` panics, then `tokio::run` will still succeed and the test will succeed.